src/wireviz/Harness.py

```diff
@@ -330,90 +351,118 @@ def output(self, filename: (str, Path), view: bool = False, cleanup: bool = True
+                gauge_name = f' x {cable.gauge} {cable.gauge_unit}' if cable.gauge else ' wires'
+                if cable.category != 'bundle':
+                    # process cable as a single entity
+                    shield_name = ' shielded' if cable.shield else ''
+                    name = f'Cable{cable_type}, {cable.wirecount}{gauge_name}{shield_name}'
+                    item = {'item': name, 'qty': cable.length, 'unit': 'm', 'designators': cable.name,
+                            'manufacturer': remove_line_breaks(cable.manufacturer), 'mpn': remove_line_breaks(cable.mpn), 'pn': cable.pn}
+                    bom_items.append(item)
+                else:
+                    # add each wire from the bundle to the bom
+                    for index, color in enumerate(cable.colors, 0):
+                        wire_color = f', {color}' if color else ''
+                        name = f'Wire{cable_type}{gauge_name}{wire_color}'
+                        item = {'item': name, 'qty': cable.length, 'unit': 'm', 'designators': cable.name,
+                                'manufacturer': remove_line_breaks(index_if_list(cable.manufacturer, index)),
+                                'mpn': remove_line_breaks(index_if_list(cable.mpn, index)), 'pn': index_if_list(cable.pn, index)}
+                        bom_items.append(item)
+
+            for part in cable.additional_components:
+                qty = part.qty * cable.get_qty_multiplier(part.qty_multiplier)
+                bom_items.append(
+                    {
+                        'item': part.description,
+                        'qty': qty,
+                        'unit': part.unit,
+                        'manufacturer': part.manufacturer,
+                        'mpn': part.mpn,
+                        'pn': part.pn,
+                        'designators': cable.name
+                    }
+                )
+
+        for item in self.additional_bom_items:
+            item = {'item': item.get('description', ''), 'qty': item.get('qty'), 'unit': item.get('unit'), 'designators': item.get('designators'),
+                    'manufacturer': item.get('manufacturer'), 'mpn': item.get('mpn'), 'pn': item.get('pn')}
+            bom_items.append(item)
+
+        # deduplicate bom
+        bom_types_group = lambda bt: (bt['item'], bt['unit'], bt['manufacturer'], bt['mpn'], bt['pn'])
+        for group in Counter([bom_types_group(v) for v in bom_items]):
+            items = [v for v in bom_items if bom_types_group(v) == group]
             shared = items[0]
-            designators = [i['designator'] for i in items]
+            designators = []
+            for item in items:
+                if item.get('designators'):
+                    if isinstance(item['designators'], List):
+                        designators.extend(item['designators'])
+                    else:
+                        designators.append(item['designators'])
```

Comment:
I suggest renaming `items` to `group_entries` and `item` to `inp` to avoid any confusion with the BOM `item` column name.